### ISSUE TITLE
"Invalid component designator" error for some systems

### ISSUE THREAD

**Issue by svetlyak40wt** [guicho271828/asdf-viz#2]
For example, running `asdf-viz deps.png spinneret` I receive the following error:

```
Making core for Roswell...
building dump:/Users/art/.roswell/impls/x86-64/darwin/sbcl-bin/1.4.12/dump/asdf-viz.core
up to date. stop
Unhandled ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                                      {10004D05B3}>:
  Invalid component designator: ((REQUIRE EXTENSIBLE-SEQUENCES))

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10004D05B3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR {1001F28FD3}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK SB-EXT:*INVOKE-DEBUGGER-HOOK* #<ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR {1001F28FD3}>)
2: (INVOKE-DEBUGGER #<ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR {1001F28FD3}>)
3: (ERROR ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR :FORMAT-CONTROL "~@<Invalid component designator: ~3i~_~A~@:>" :FORMAT-ARGUMENTS (((:REQUIRE :EXTENSIBLE-SEQUENCES))))
4: (ASDF/SESSION:SYSDEF-ERROR "~@<Invalid component designator: ~3i~_~A~@:>" ((:REQUIRE :EXTENSIBLE-SEQUENCES)))
5: ((:METHOD ASDF/SYSTEM:FIND-SYSTEM (T)) ((:REQUIRE :EXTENSIBLE-SEQUENCES)) T) [fast-method]
6: ((:METHOD GRAPH-OBJECT-POINTS-TO ((EQL (QUOTE DEPENDSON)) ASDF/SYSTEM:SYSTEM)) #<unused argument> #<ASDF/SYSTEM:SYSTEM "serapeum">) [fast-method]
7: ((LABELS CL-DOT::HANDLE-OBJECT :IN CL-DOT::CONSTRUCT-GRAPH) #<ASDF/SYSTEM:SYSTEM "serapeum">)
8: (SB-KERNEL:%MAP-FOR-EFFECT-ARITY-1 #<CLOSURE (LABELS CL-DOT::HANDLE-OBJECT :IN CL-DOT::CONSTRUCT-GRAPH) {1004B1F29B}> (#<ASDF/SYSTEM:SYSTEM "parenscript"> #<ASDF/SYSTEM:SYSTEM "alexandria"> #<ASDF/SYSTEM:SYSTEM "cl-ppcre"> #<ASDF/SYSTEM:SYSTEM "global-vars"> #<ASDF/SYSTEM:SYSTEM "serapeum"> #<ASDF/SYSTEM:SYSTEM "trivial-gray-streams">))
9: ((LABELS CL-DOT::HANDLE-OBJECT :IN CL-DOT::CONSTRUCT-GRAPH) #<ASDF/SYSTEM:SYSTEM "spinneret">)
10: (SB-KERNEL:%MAP-FOR-EFFECT-ARITY-1 #<CLOSURE (LABELS CL-DOT::HANDLE-OBJECT :IN CL-DOT::CONSTRUCT-GRAPH) {1004B1F29B}> (#<ASDF/SYSTEM:SYSTEM "spinneret">))
11: (CL-DOT::CONSTRUCT-GRAPH DEPENDSON (#<ASDF/SYSTEM:SYSTEM "spinneret">))
12: ((:METHOD GENERATE-GRAPH-FROM-ROOTS (T T)) DEPENDSON (#<ASDF/SYSTEM:SYSTEM "spinneret">) (:RANKDIR "LR")) [fast-method]
13: (VISUALIZE-ASDF-HIERARCHY "deps.png" (#<ASDF/SYSTEM:SYSTEM "spinneret">) DEPENDSON)
14: (MAIN "deps.png" "spinneret")
15: (SB-INT:SIMPLE-EVAL-IN-LEXENV (APPLY (QUOTE MAIN) ROSWELL:*ARGV*) #<NULL-LEXENV>)
16: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) #<NULL-LEXENV>)
17: (SB-EXT:EVAL-TLF (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL NIL)
18: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL)
19: (SB-INT:LOAD-AS-SOURCE #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
20: ((FLET SB-FASL::THUNK :IN LOAD))
21: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<CLOSURE (FLET SB-FASL::THUNK :IN LOAD) {19FF52B}> #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}>)
22: ((FLET SB-FASL::LOAD-STREAM :IN LOAD) #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}> NIL)
23: (LOAD #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}> :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
24: ((FLET ROSWELL::BODY :IN ROSWELL:SCRIPT) #<SB-SYS:FD-STREAM for "file /Users/art/.roswell/bin/asdf-viz" {1003B1B763}>)
25: (ROSWELL:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "spinneret")
26: (ROSWELL:RUN ((:EVAL "(ros:asdf)") (:EVAL "(ros:quicklisp)") (:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "spinneret") (:QUIT NIL)))
27: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:RUN (QUOTE ((:EVAL "(ros:asdf)") (:EVAL "(ros:quicklisp)") (:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "spinneret") (:QUIT NIL)))) #<NULL-LEXENV>)
28: (EVAL (ROSWELL:RUN (QUOTE ((:EVAL "(ros:asdf)") (:EVAL "(ros:quicklisp)") (:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "spinneret") (:QUIT NIL)))))
29: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(progn #-ros.init(cl:load \"/Users/art/usr/roswell/etc/roswell/init.lisp\"))") (:EVAL . "(ros:run '((:eval\"(ros:asdf)\")(:eval\"(ros:quicklisp)\")(:script \"/Users/art/.roswell/bin/asdf-viz\"\"deps.png\"\"spinneret\")(:quit ())))")))
30: (SB-IMPL::TOPLEVEL-INIT)
31: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
32: ((FLET "WITHOUT-INTERRUPTS-BODY-34" :IN SB-EXT:SAVE-LISP-AND-DIE))
33: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
```

And when running the `asdf-viz deps.png dexador`:

```
Making core for Roswell...
building dump:/Users/art/.roswell/impls/x86-64/darwin/sbcl-bin/1.4.12/dump/asdf-viz.core
up to date. stop
Unhandled ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                                      {10004D05B3}>:
  Invalid component designator: (cl+ssl)

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10004D05B3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR {1004B1D9E3}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK SB-EXT:*INVOKE-DEBUGGER-HOOK* #<ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR {1004B1D9E3}>)
2: (INVOKE-DEBUGGER #<ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR {1004B1D9E3}>)
3: (ERROR ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR :FORMAT-CONTROL "~@<Invalid component designator: ~3i~_~A~@:>" :FORMAT-ARGUMENTS (("cl+ssl")))
4: (ASDF/SESSION:SYSDEF-ERROR "~@<Invalid component designator: ~3i~_~A~@:>" ("cl+ssl"))
5: ((:METHOD ASDF/SYSTEM:FIND-SYSTEM (T)) ("cl+ssl") T) [fast-method]
6: ((:METHOD GRAPH-OBJECT-POINTS-TO ((EQL (QUOTE DEPENDSON)) ASDF/SYSTEM:SYSTEM)) #<unused argument> #<ASDF/SYSTEM:SYSTEM "dexador">) [fast-method]
7: ((LABELS CL-DOT::HANDLE-OBJECT :IN CL-DOT::CONSTRUCT-GRAPH) #<ASDF/SYSTEM:SYSTEM "dexador">)
8: (SB-KERNEL:%MAP-FOR-EFFECT-ARITY-1 #<CLOSURE (LABELS CL-DOT::HANDLE-OBJECT :IN CL-DOT::CONSTRUCT-GRAPH) {1004462D2B}> (#<ASDF/SYSTEM:SYSTEM "dexador">))
9: (CL-DOT::CONSTRUCT-GRAPH DEPENDSON (#<ASDF/SYSTEM:SYSTEM "dexador">))
10: ((:METHOD GENERATE-GRAPH-FROM-ROOTS (T T)) DEPENDSON (#<ASDF/SYSTEM:SYSTEM "dexador">) (:RANKDIR "LR")) [fast-method]
11: (VISUALIZE-ASDF-HIERARCHY "deps.png" (#<ASDF/SYSTEM:SYSTEM "dexador">) DEPENDSON)
12: (MAIN "deps.png" "dexador")
13: (SB-INT:SIMPLE-EVAL-IN-LEXENV (APPLY (QUOTE MAIN) ROSWELL:*ARGV*) #<NULL-LEXENV>)
14: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) #<NULL-LEXENV>)
15: (SB-EXT:EVAL-TLF (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL NIL)
16: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL)
17: (SB-INT:LOAD-AS-SOURCE #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
18: ((FLET SB-FASL::THUNK :IN LOAD))
19: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<CLOSURE (FLET SB-FASL::THUNK :IN LOAD) {19FF52B}> #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}>)
20: ((FLET SB-FASL::LOAD-STREAM :IN LOAD) #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}> NIL)
21: (LOAD #<CONCATENATED-STREAM :STREAMS NIL {1003B1D653}> :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
22: ((FLET ROSWELL::BODY :IN ROSWELL:SCRIPT) #<SB-SYS:FD-STREAM for "file /Users/art/.roswell/bin/asdf-viz" {1003B1B763}>)
23: (ROSWELL:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "dexador")
24: (ROSWELL:RUN ((:EVAL "(ros:asdf)") (:EVAL "(ros:quicklisp)") (:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "dexador") (:QUIT NIL)))
25: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:RUN (QUOTE ((:EVAL "(ros:asdf)") (:EVAL "(ros:quicklisp)") (:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "dexador") (:QUIT NIL)))) #<NULL-LEXENV>)
26: (EVAL (ROSWELL:RUN (QUOTE ((:EVAL "(ros:asdf)") (:EVAL "(ros:quicklisp)") (:SCRIPT "/Users/art/.roswell/bin/asdf-viz" "deps.png" "dexador") (:QUIT NIL)))))
27: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(progn #-ros.init(cl:load \"/Users/art/usr/roswell/etc/roswell/init.lisp\"))") (:EVAL . "(ros:run '((:eval\"(ros:asdf)\")(:eval\"(ros:quicklisp)\")(:script \"/Users/art/.roswell/bin/asdf-viz\"\"deps.png\"\"dexador\")(:quit ())))")))
28: (SB-IMPL::TOPLEVEL-INIT)
29: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
30: ((FLET "WITHOUT-INTERRUPTS-BODY-34" :IN SB-EXT:SAVE-LISP-AND-DIE))
31: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
```

**Comment by guicho271828** [comment]
Oh, this is a degradation. `:require` used to work.

**Comment by guicho271828** [comment]
related: #1 (I don't have enough adrenaline to fix this, maybe some time)

**Comment by guicho271828** [comment]
I tested it today, but could not reproduce either behavior.
See https://github.com/guicho271828/asdf-viz/blob/master/spinneret.png

I'm on sbcl-bin 1.4.10 on ubuntu linux. Will test it on 1.5.0

**Comment by guicho271828** [comment]
It worked on 1.5.0, as well as on ccl 1.11.5 . I suspect you forgot updating. The issue should have been fixed in #1 already.

**Comment by svetlyak40wt** [comment]
Thank you. I'll try to update.
